astroquery/mast/core.py

```diff
@@ -1304,7 +1339,69 @@ def _download_files(self, products, base_dir, cache=True):
             url = None
 
             try:
-                self._download_file(dataUrl, localPath, cache=cache)
+                s3_fetch_ok = False
+                if boto3 is not None and dataUri.startswith("mast:HST/product"):
+                    # The following is a mishmash of BaseQuery._download_file and s3 access through boto
+
+                    try:
+                        obs_id = obs_id.lower()
+
+                        # magic associations logic per Brian - 0-9/a-e we convert to 0
+                        magicValues = "123456789abcde"
+                        if obs_id[-1] in magicValues:
+                            print("WARNING: IPPPSSOOT Magic in %s" % dataUri)
+                            new_obs_id = obs_id[:-1] + "0"
+                            dataUri = dataUri.replace(obs_id, new_obs_id)
+                            obs_id = new_obs_id
+                            print("WARNING: New URI %s" % dataUri)
+
+                        bucketPath = "hst/public/" + obs_id[:4] + dataUri.replace("mast:HST/product", "")
+
+                        # CAOM LIES!
```

Comment:
I like the comment.  What's CAOM?

Reply:
It's the backing database for the mast services which populate the data passed to this function.

Somehow we have incorrect filesizes reported through that database which don't match up with the files in our archive...